website/client/src/components/inventory/equipment/attributesPopover.vue

```diff
@@ -5,7 +5,7 @@
         {{ `${$t('lockedItem')}` }}
       </h4>
       <div
-        v-if="item.specialClass"
+        v-if="item.specialClass !== user.stats.class && item.klass !== user.stats.class"
```

Comment:
Hi @laurel-thomson ! Should the check here be different in case the item doesn't have the `specialClass`
property? Also i think all items with a special class have `item.klass` set to `special` which will never match the user's own class. Here it's how it's checked on the server side https://github.com/HabitRPG/habitica/blob/bd8e67a2ea3d6109ab2e43057ecdfe8f1be2e363/website/common/script/ops/buy/buyMarketGear.js#L27-L28 (btw, if it's possible it would be great if the check could be abstracted an reused)

cc'ing @Alys since she knows more about this

Reply:
> Should the check here be different in case the item doesn't have the specialClass property?

Yes, there'd need to be a check for specialClass existing. @laurel-thomson in case it helps, here's an example of an item that just has klass, taken from the output of the API's `content` route:

"weapon_warrior_0":{
   "text":"Training Sword","notes":"Practice...","value":1,"type":"weapon","key":"weapon_warrior_0","set":"warrior-0",
   **"klass":"warrior",**
   "index":"0","str":0,"int":0,"per":0,"con":0
}

> Also i think all items with a special class have item.klass set to special which will never match the user's own class.

Yes. Here's an example of a special class item:

"weapon_special_spring2020Warrior": {
   "event":{"start":"2019-03-17","end":"2020-05-02"},
   **"specialClass":"warrior",**
   "set":"spring2020BeetleWarriorSet","text":"Sharpened Wing","notes":"Fight...","value":90,"str":15,"type":"weapon","key":"weapon_special_spring2020Warrior",
   **"klass":"special",**
   "index":"spring2020Warrior","int":0,"per":0,"con":0
},

The code that paglias referenced looks good for checking klass and specialClass, although it would be clearer if `checkKlass` was renamed to `wrongKlass` and `checkSpecialClass` was renamed to `wrongSpecialClass`:
https://github.com/HabitRPG/habitica/blob/bd8e67a2ea3d6109ab2e43057ecdfe8f1be2e363/website/common/script/ops/buy/buyMarketGear.js#L24-L34


However from looking at that code and then doing a test, I've realised that the API doesn't prevent you buying a "tierLocked" item. I.e., if you don't own either of these:
- armor_warrior_2
- armor_warrior_4

then you should be allowed to buy armor_warrior_2 but not armor_warrior_4. The API DOES allow you to buy armor_warrior_4 before armor_warrior_2 using the command below - that's a bug. 
`curl -X POST -d "" -H "x-api-user: d904bd62-da08-416b-a816-ba797c9ee265" -H "x-api-key: ......." https://habitica.com/api/v3/user/buy-gear/armor_warrior_4`


I'm sorry @laurel-thomson it's a bit of a complex situation and we didn't realise that the API had a different bug than the website. :-/ If you'd like to keep working on this that would be awesome and we'll be happy to answer any questions you have!

Reply:
Thanks for all the information!  I can keep working on this.